### ISSUE TITLE
GCS registry driver not working

### ISSUE THREAD

**Issue by bdurrow** [openshift/origin#10659]
#10640 doesn't appear to fix my problem.  I added comments about how I was able to get it working in that PR.

**Comment by mfojtik** [comment]
@bdurrow I just talked to the QA guys and they were able to verify this fix (they are able to push images into GCS backend).

**Comment by mfojtik** [comment]
@bdurrow are you using 2.4.0 registry (the one we have in `vendor/github.com/docker/distribution/registry` ?

**Comment by mfojtik** [comment]
@bdurrow I believe your fix is right as mentioned in the original pull request and it fixes the "root cause" of this. However, the upstream PR is meant to be for Docker Registry 2.5.0 where we use Docker Registry 2.4.0 internally. I'm not sure why the the https://github.com/openshift/origin/pull/10640 does not fix the problem with pushing for you, but it should address this issue for 2.4.0 registry. Once we bump the version to 2.5.0, we will get the fix you propose.

**Comment by bdurrow** [comment]
As I just mentioned in #10640 I tried with the image that I assume included this fix and I still got an error (Not the same error as before the fix).  I tried to only pull from upstream PRs that were compatible for 2.4.0.  Upstream has an unmerged PR against 2.4.0 but it has been polluted with other commits when the submitter pulled master into their feature branch.  My environment is made from release-1.2 with the exception of the registry which now works with the origin-docker-registry image I made using the method described.  I know that mixing and matching like this is not supported but the risk seems smaller to do it that way than using 1.3.0-alpha for all the components.
